### ISSUE TITLE
@decorators.require_login not work

### ISSUE THREAD

**Issue by zhangchunlin** [limodou/uliweb3#5]
decorators.require_login cannot be used as decorator
e.g.:
```
@expose('/user')
class UserView(object):
    @decorators.require_login
    def view(self):
        return UsersManageView().view(request.user.id)
```
When access, will get error like:
```
[INFO] 127.0.0.1 - - [29/Nov/2018 17:34:38] "GET /user/view HTTP/1.1" 500 -
[ERROR] Error on request:
Traceback (most recent call last):
  File "/home/zhangclb/oss/uliweb3/uliweb/lib/werkzeug/debug/__init__.py", line 101, in __call__
    appiter = self.application(environ, start_response)
  File "/home/zhangclb/oss/uliweb3/uliweb/contrib/staticfiles/wsgi_staticfiles.py", line 91, in __call__
    return self.app(environ, start_response)
  File "/home/zhangclb/oss/uliweb3/uliweb/core/SimpleFrame.py", line 1541, in __call__
    response = self._open(environ)
  File "/home/zhangclb/oss/uliweb3/uliweb/core/SimpleFrame.py", line 1464, in _open
    mod, handler_cls, handler = self.prepare_request(req, rule)
  File "/home/zhangclb/oss/uliweb3/uliweb/core/SimpleFrame.py", line 960, in prepare_request
    _klass, mod, handler = self.get_handler(endpoint)
  File "/home/zhangclb/oss/uliweb3/uliweb/core/SimpleFrame.py", line 946, in get_handler
    _klass = get_class(handler)()
  File "/home/zhangclb/oss/uliweb3/uliweb/utils/_compat.py", line 95, in get_class
    meth.__qualname__.split('.<locals>', 1)[0].rsplit('.', 1)[0])
AttributeError: module 'uliweb_lapps.auth.user_admin.views_admin' has no attribute 'wraps'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/zhangclb/oss/uliweb3/uliweb/lib/werkzeug/serving.py", line 177, in run_wsgi
    execute(self.server.app)
  File "/home/zhangclb/oss/uliweb3/uliweb/lib/werkzeug/serving.py", line 167, in execute
    for data in application_iter:
  File "/home/zhangclb/oss/uliweb3/uliweb/lib/werkzeug/debug/__init__.py", line 115, in __call__
    debug_context = self.create_debug_context(environ, exc_info)
  File "/home/zhangclb/oss/uliweb3/uliweb/lib/werkzeug/debug/__init__.py", line 153, in create_debug_context
    frame = get_frame_info(tb, simple=simple)
  File "/home/zhangclb/oss/uliweb3/uliweb/lib/werkzeug/debug/util.py", line 315, in get_frame_info
    parsed_source = highlight_python(source)
  File "/home/zhangclb/oss/uliweb3/uliweb/lib/werkzeug/debug/util.py", line 139, in highlight_python
    parser.parse()
  File "/home/zhangclb/oss/uliweb3/uliweb/lib/werkzeug/debug/util.py", line 188, in parse
    for v in tokenize.generate_tokens(text.readline):
  File "/home/zhangclb/.local/env3/lib/python3.6/tokenize.py", line 558, in _tokenize
    if line[pos] in '#\r\n':           # skip comments or blank lines
TypeError: 'in <string>' requires string as left operand, not int

```

**Comment by zhangchunlin** [comment]
If I modify in "def require_login" from 
```
    @wraps(f)
    def _f(*args, **kwargs):
        _login(next=next)
        return f(*args, **kwargs)
    return _f
```
to
```
    #@wraps(f)
    def _f(*args, **kwargs):
        _login(next=next)
        return f(*args, **kwargs)
    func = wraps(f)
    print(func,f)
    _f = func(_f)
    return _f
```
still get the same error

The print result is
```
<function wraps.<locals>._f at 0x7f8a74a5b598> <function UserView.view at 0x7f8a74a5b400>
```

is it connect to the "wraps.<locals>._f"?

**Comment by limodou** [comment]
Fixed d8b739de19e325c1a35fdf1bd803071bc6f96e0f
